### PR TITLE
MI-28 disconnected networks

### PR DESCRIPTION
**Pull request by JanionNevill**
This adds disconnected network detection to both the uploaded data, and the sensitivity analysis. It automatically excludes studies which are disconnected to the primary network (that which contains the reference treatment).

The reactivity chain looks like this:
image

### CODE REVIEW

**Comment by JanionNevill** [comment]
Hi @nabury, do you have any estimate on when you might get to this?

**Review by nabury — COMMENTED** [review]
Seems to work ok when testing manually and also automated tests.

I made a couple of other minor questions on the review.

When testing manually I noticed that the uploaded file 'disappeared' when uploading a disconnected network 

Screenshot 2024-08-07 at 16 01 51

I managed to do something when testing that gave me a text warning in section 2a results with selected studies excluded that was something about the reference treatment being different to the one without studies excluded. I didn't get a screenshot and haven't been able to reproduce this.

ETA - the way this works is nice and easy to understand for the user

**Comment by JanionNevill** [comment]
> When testing manually I noticed that the uploaded file 'disappeared' when uploading a disconnected network

I can't work out why this is happening. I tried debugging on all the places where the upload input is replaced, but it didn't trigger any of them. I think it's not a big enough issue to hold this back though. what do you think?

**Comment by JanionNevill** [comment]
> I managed to do something when testing that gave me a text warning in section 2a results with selected studies excluded that was something about the reference treatment being different to the one without studies excluded. I didn't get a screenshot and haven't been able to reproduce this.

That's a feature which has been in the app for ages apparently, not something I added for this task.

File: R/sub_networks.R [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+#'
+#' @return List of subnetworks, where each subnetwork is a list containing:
+#' - "treatments" = The IDs of the treatments included in the given network
+#' - "studies" = The names of the studies included in the given subnetwork
+IdentifySubNetworks <- function(data, treatment_df, reference_treatment_name = NULL, subnet_name_prefix = "subnet_") {
+  if (is.null(reference_treatment_name)) {
+    reference_treatment <- 1
+  } else if (!(reference_treatment_name %in% treatment_df$Label)) {
+    # Warn users if reference treatment isn't contained within the data
+    warning(glue::glue("Reference treatment '{reference_treatment_name}' cannot be found in the data. ",
```
[the remark this answers is not in the document]
Comment (JanionNevill) [reply R/sub_networks.R:38]: Tests on line 234 of `test-sub_networks.R` trigger this.

File: R/data_analysis_page.R [hunk 2]
```diff
@@ -74,48 +72,86 @@ data_analysis_page_server <- function(id, data, is_default_data, treatment_df, m
...
+    sensitivity_data = analysis_options_reactives$sensitivity_data
+    sensitivity_treatment_list = analysis_options_reactives$sensitivity_treatment_list
     rank_option = analysis_options_reactives$rank_option
     continuous_outcome = analysis_options_reactives$continuous_outcome
     binary_outcome = analysis_options_reactives$binary_outcome
 
     #####
     # Reactive functions used in various places, based on the data
     #####
+
```
[the remark this answers is not in the document]
Comment (JanionNevill) [reply R/data_analysis_page.R:92]: Done

**Review by nabury — APPROVED** [review]

**Pull request merged**